llama-index-integrations/indices/llama-index-indices-managed-postgresml/llama_index/indices/managed/postgresml/base.py

```diff
@@ -0,0 +1,198 @@
+"""Managed index.
+
+A managed Index - where the index is accessible via some API that
+interfaces a managed service.
+
+"""
+
+import logging
+from llama_index.core.async_utils import run_async_tasks
+from typing import Any, Dict, Optional, Sequence, Type
+
+from llama_index.core.base.base_query_engine import BaseQueryEngine
+from llama_index.core.base.base_retriever import BaseRetriever
+from llama_index.core.data_structs.data_structs import IndexDict, IndexStructType
+from llama_index.core.indices.managed.base import BaseManagedIndex, IndexType
+from llama_index.core.schema import (
+    BaseNode,
+    Document,
+    TextNode,
+)
+from pgml import Collection, Pipeline
+
+_logger = logging.getLogger(__name__)
+
+
+class PostgresMLIndexStruct(IndexDict):
+    """PostgresML Index Struct."""
+
+    @classmethod
+    def get_type(cls) -> IndexStructType:
+        """Get index struct type."""
+        return IndexStructType.POSTGRESML
```

Comment:
The index struct is so silly 😅 Since this depends on a change in core, can we hardcode the string here instead until core gets published?